src/bash.rs

```diff
@@ -88,7 +88,7 @@ pub fn write_completion_script<W: Write>(buffer: &mut W, command: &str, dfa: &DF
             local word=${{COMP_WORDS[$word_index]}}
             local word_matched=0
             for literal_id in $(seq 1 ${{#literals[@]}}); do
-                if [[ ${{literals[$literal_id]}} = $word ]]; then
+                if [[ ${{literals[$literal_id]}} = "$word" ]]; then
```

Comment:
The double quotes within `[[` shouldn’t be necessary and I tend to omit them for readability.   This works for instance:

```
$ touch 'foo bar'
$ x="foo bar"
$ [[ -e $x ]] && echo yes
yes
```

Reply:
It still performs globbing, see [shellcheck wiki](https://github.com/koalaman/shellcheck/wiki/SC2053).

Reply:
But if you don't like the quotes, I will remove them from this PR.

Reply:
> But if you don't like the quotes, I will remove them from this PR.
No no, you’re totally right.  The quotes are the right way here.  What I’m not on board with is the other changes that stringify the array.